### PR TITLE
Remove SQLite production warning

### PR DESCRIPTION
**Pull request by intrip**
### Motivation / Background

SQLite is becoming a more capable database and there are more companies using it in production. It's really good for self-hosted/small environments where you don't want to boot a separate server for the DB. There's no more need to discourage his usage via a warning.

See https://github.com/rails/rails/pull/49633

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by zzak** [comment]
You created this monster :joy: #42191

**Comment by intrip** [comment]
> You created this monster 😂 #42191

Ahah yes! Time to kill it? 😅

File: activerecord/lib/active_record/railtie.rb [hunk 1]
```diff
@@ -230,16 +229,6 @@ class Railtie < Rails::Railtie # :nodoc:
...
     end
 
-    SQLITE3_PRODUCTION_WARN = "You are running SQLite in production, this is generally not recommended."\
-      " You can disable this warning by setting \"config.active_record.sqlite3_production_warning=false\"."
-    initializer "active_record.sqlite3_production_warning" do
-      if config.active_record.sqlite3_production_warning && Rails.env.production?
-        ActiveSupport.on_load(:active_record_sqlite3adapter) do
-          Rails.logger.warn(SQLITE3_PRODUCTION_WARN)
-        end
-      end
```
Comment (skipkayhil) [inline activerecord/lib/active_record/railtie.rb:240]: Should we replace this with a deprecation warning if someone sets the value to `false`? Otherwise I worry apps will end up with a configuration line that does nothing left in their applications

Comment (intrip) [reply activerecord/lib/active_record/railtie.rb:240]: Good point, let's do it. 
I'll also work on the [docs addition](https://github.com/rails/rails/pull/49633#issuecomment-1773034807)

**Review by jhawthorn — APPROVED** [review]

**Comment by byroot** [comment]
So coincidentally this was the subject of discussion at Rails World, and to summarize

  - The warning is probably a bit too scary.
  - It doesn't really explain why it's even there.

However:

  - Platforms with ephemeral file system are still extremely common deployment targets especially for newcomers

So we agreed to reword it to be more helpful, but not to remove it.

**Comment by byroot** [comment]
Another discussed alternative is to remove the warnings, but to generate the `database.yml` with the production database not fully configured, with a comment explaining that the DB must be in a persistent path (e.g. Docker mounted volume or whatever)

**Comment by intrip** [comment]
> Another discussed alternative is to remove the warnings, but to generate the `database.yml` with the production database not fully configured, with a comment explaining that the DB must be in a persistent path (e.g. Docker mounted volume or whatever)

Thanks @byroot! I like that approach, I'll keep this PR open and extend it accordingly then.
